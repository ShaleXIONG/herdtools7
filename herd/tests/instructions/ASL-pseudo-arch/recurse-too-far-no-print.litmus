ASL recurse-too-far-no-print

{ }

func T0(x:integer) => integer
recurselimit 10
begin
  if x <= 0 then return 0;
  else return 1+T0(x-1);
  end;
end;

func main() => integer
begin
  let n = T0(10);
  print(123456);
  return 0;
end;

locations [0:main.0.n;]
