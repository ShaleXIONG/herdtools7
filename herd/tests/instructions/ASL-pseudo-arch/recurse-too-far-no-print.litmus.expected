Test recurse-too-far-no-print Required
States 0
Loop Ok
Witnesses
Positive: 0 Negative: 0
Condition forall (true)
Observation recurse-too-far-no-print Never 0 0
Hash=abd5196ee6a439b4baf00eefdc7b590e

